tomcat/src/main/java/org/apache/coyote/http11/HttpRequestMessageParser.java

```diff
@@ -0,0 +1,48 @@
+package org.apache.coyote.http11;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class HttpRequestMessageParser {
+
+    private final BufferedReader bufferedReader;
+
+    public HttpRequestMessageParser(final BufferedReader bufferedReader) {
+        this.bufferedReader = bufferedReader;
+    }
+
+    public HttpRequestMessage parseRequestMessage() throws IOException {
+        String startLine = bufferedReader.readLine();
+        if (startLine == null || startLine.isBlank()) {
+            return new HttpRequestMessage(null, null, null);
+        }
+
+        List<String> headers = new ArrayList<>();
+        for (String data = bufferedReader.readLine(); data != null; data = bufferedReader.readLine()) {
+            if (data.isBlank()) {
+                break;
+            }
+            headers.add(data);
+        }
+
+        Map<String, String> headerKeyValue = new HashMap<>();
+        for (int i = 0; i < headers.size(); i++) {
+            final String[] keyValue = headers.get(i).split(": ");
+            headerKeyValue.put(keyValue[0], keyValue[1]);
+        }
+
+        String body = null;
+        if (startLine.startsWith("POST")) {
```

Comment:
`body` 에 데이터를 넣을지 결정하는 것은 `HTTP Method` 의 `POST` 여부로 판별하기 보다는 `Content-Length` 의 여부로 판별하는 것이 더 적절한 방법이 될 것 같아요~

> https://httpwg.org/specs/rfc9110.html#field.content-length
Because Content-Length is used for **message delimitation** in HTTP/1.1, its field value can impact how the message is parsed by downstream recipients even when the immediate connection is not using HTTP/1.1. If the message is forwarded by a downstream intermediary, a Content-Length field value that is inconsistent with the received message framing might cause a security failure due to request smuggling or response splitting.<br>
As a result, a sender MUST NOT forward a message with a Content-Length header field value that is known to be incorrect.

만약, `Content-Length` 와 실제 받은 데이터의 길이가 다를 경우 어떤 정책을 세울지 고민을 해봐도 좋을 것 같아요!

Reply:
새양의 의견에 동의합니다!
`Content-Length` 의 여부로 판별하는 방식으로 수정하였습니다.

Content-Length 와 실제 받은 데이터의 길이가 다를 경우 어떤 정책을 세울지 고민해 보았는데요,
- Content-Length가 더 길 경우: 408 Request Timeout 응답
- Content-Length가 더 짧을 경우: body 데이터를 Content-Length 값 길이로 자른다.

저라면 이렇게 대응할 것 같습니다!